core/editor/src/tools/mod.rs

```diff
@@ -1,14 +1,32 @@
-use crate::events::{ModKeys, MouseState, TracePoint};
-use crate::{events::Trace, Color};
+mod crop;
+mod ellipse;
+mod line;
+mod navigate;
+mod path;
+mod pen;
+mod rectangle;
+mod sample;
+mod select;
+mod shape;
+
+use crate::events::{Event, ModKeys, MouseState, Trace, TracePoint};
+use crate::Color;
+use crate::EditorError;
+use document_core::Operation;
 use std::collections::HashMap;
 
+pub trait Tool {
+	fn handle_input(&mut self, event: Event) -> Option<Operation>;
```

Comment:
We'll want to replace `Option` with `Vec` so a queue of `Operation`s can be returned.